sdk/log/simple.go

```diff
@@ -55,6 +94,22 @@ func (s *SimpleProcessor) OnEmit(ctx context.Context, r *Record) error {
 		simpleProcRecordsPool.Put(records)
 	}()
 
+	defer func() {
+		if s.selfObservabilityEnabled {
```

Comment:
I prefer placing `defer func() {` inside the if statement to minimize unnecessary performance overhead when it's not enabled.

Reply:
Done!